Services/src/main/java/com/agilepro/services/admin/ProjectMemberService.java

```diff
@@ -0,0 +1,102 @@
+package com.agilepro.services.admin;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.agilepro.commons.models.admin.EmployeeModel;
+import com.agilepro.commons.models.customer.ProjectMemberModel;
+import com.agilepro.persistence.entity.admin.ProjectMemberEntity;
+import com.agilepro.persistence.repository.admin.IProjectMemberRepository;
+import com.yukthi.persistence.ITransaction;
+import com.yukthi.persistence.repository.RepositoryFactory;
+import com.yukthi.utils.exceptions.InvalidStateException;
+import com.yukthi.webutils.services.BaseCrudService;
+
+/**
+ * The Class ProjectMemberService.
+ * 
+ * @author Pritam
+ */
+@Service
+public class ProjectMemberService extends BaseCrudService<ProjectMemberEntity, IProjectMemberRepository>
+{
+	/**
+	 * The repository factory.
+	 **/
+	@Autowired
+	private RepositoryFactory repositoryFactory;
+
+	/**
+	 * The employee service.
+	 **/
+	@Autowired
+	private EmployeeService employeeService;
+
+	/**
+	 * Instantiates a new project members service.
+	 */
+	public ProjectMemberService()
+	{
+		super(ProjectMemberEntity.class, IProjectMemberRepository.class);
+	}
+
+	/**
+	 * Fetch project members.
+	 *
+	 * @param projectId
+	 *            the project id
+	 * @return the list
+	 */
+	public List<ProjectMemberModel> fetchProjectMembers(Long projectId)
+	{
+		IProjectMemberRepository iprojectMemberRepository = repositoryFactory.getRepository(IProjectMemberRepository.class);
+
+		List<ProjectMemberEntity> projectMemberEntities = iprojectMemberRepository.fetchProjMemByProjId(projectId);
+
+		List<ProjectMemberModel> projectMemberModels = new ArrayList<>();
+
+		ProjectMemberModel projectMemberModel;
+
+		EmployeeModel employeeModel;
+
+		for(ProjectMemberEntity pentity : projectMemberEntities)
+		{
+			projectMemberModel = super.toModel(pentity, ProjectMemberModel.class);
+
+			employeeModel = employeeService.fetchEmployee(projectMemberModel.getEmployeeId());
+
+			projectMemberModel.setPhoto(employeeModel.getPhoto());
+			projectMemberModel.setName(employeeModel.getName());
+
+			projectMemberModels.add(projectMemberModel);
+		}
+		return projectMemberModels;
+	}
+
+	/**
+	 * Delete by employee id.
+	 *
+	 * @param id
+	 *            the id
+	 */
+	public void deleteByEmployeeId(Long id)
+	{
+		try(ITransaction transaction = repository.newOrExistingTransaction())
+		{
+			IProjectMemberRepository iprojectMemberRepository = repositoryFactory.getRepository(IProjectMemberRepository.class);

```

Comment:
Why are we getting repository here it should be at class level like other repos.